vendor/blockchain_contracts/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+#![warn(unused_extern_crates, missing_debug_implementations, rust_2018_idioms)]
+#![deny(unsafe_code)]
+
+#[macro_use]
+extern crate serde;
+
+use std::cmp::Ordering;
+
+pub mod ethereum;
+pub mod rfc003;
+
+#[derive(Clone, Copy, Debug, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum DataName {
+    SecretHash,
+    Expiry,
+    RedeemIdentity,
+    RefundIdentity,
+    TokenQuantity,
+    TokenContract,
+}
+
+impl DataName {
+    fn to_markdown(&self) -> String {
+        serde_json::to_string(&self)
```

Comment:
❌ It's a bit too weird to use`serde_json` to do this just to get a `snake_case` name for it. Use https://docs.rs/strum/0.15.0/strum/ instead.